v0/util/index.js

```diff
@@ -829,19 +829,20 @@ const handleMetadataForValue = (value, metadata, integrationsObj = null) => {
 // It'll look for the canonical names for that integration and return the
 // `integrations` object for that destination, else null
 const getIntegrationsObj = (message, destinationName = null) => {
-  if (destinationName) {
-    const canonicalNames = DestCanonicalNames[destinationName];
-    for (let index = 0; index < canonicalNames.length; index += 1) {
-      const integrationsObj = get(
-        message,
-        `integrations.${canonicalNames[index]}`
-      );
-      if (integrationsObj) {
-        return integrationsObj;
-      }
+  if (!destinationName) {
+    return {};
```

Comment:
Got it. Though I don't agree with that pattern. We can return always two values, i.e. a `boolean` and an `object`.
I understand that this requires a more aggressive refactoring but it will eliminate implicit inferences.
In other words `getIntegrationsObj` features some hidden logic that should not be part of what the function is about to do.

Reply:
@mcchran for keeping the backward comaptibility to gracefully migrate. For cases where  we are not able to get `integrationsObj` we can retun `false` that we respect the function contract aswell

Reply:
I do not aggree with that pattern.

An easy way to return two arguments, a state and an object ...